File: Strings/__1000__/pagina01/1277/1277.js
const fs = require("fs")
const inputs = fs.readFileSync("/dev/stdin", "utf8").split("\n")
const numero_casos = Number(inputs.shift())

inputs.pop()

const calcular_frequencia = (total, faltas) => {
    const presencas = total - faltas
    const frequencia = (presencas * 100) / total
    return frequencia >= 75
}

for(let i = 0; i < numero_casos; i++){
    const numero_alunos = Number(inputs.shift())
    const alunos = inputs.shift().split(" ")
    const frenqu = inputs.shift().split(" ")
    
    let alunos_baixa_frequencia = []
    
    for(let j = 0; j < numero_alunos; j++){
        const aluno = alunos[j]
        const frequencias = frenqu[j]
        
        let faltas = 0
        let total = 0
        
        for(const frequencia of frequencias){
            if (frequencia == "A") faltas += 1
            if (frequencia == "M") continue
            total += 1
        }
        
        if (!calcular_frequencia(total, faltas)) alunos_baixa_frequencia.push(aluno)
    }
    
    console.log(alunos_baixa_frequencia.length > 0 ? alunos_baixa_frequencia.join(" ") : "")
}

File: Strings/__1000__/pagina01/1277/1277.py
#dado o nome dos alunos e suas frequências registradas
#P para presente, A para ausente e M para justificada
#calcular a frequência de presença do aulo, caso abaixo
#de 75% adicionar o nome na lista de frequencias baixas.
#printar os nomes linha a linha, caso não houver nomes
#printar uma linha em branco.

def calcular_frequencia(total, faltas):
    presencas = total - faltas
    frequencia = (presencas * 100) / total
    return frequencia >= 75

numero_casos = int(input())
for _ in range(numero_casos):
    n = int(input()) #numero de alunos
    
    alunos = [aluno for aluno in input().split()]
    freqnc = [frequ for frequ in input().split()]
    
    alunos_freq_baixa = []
    
    for i in range(n):
        aluno = alunos[i]
        frequencias = freqnc[i]
        
        faltas = 0
        total = 0
        for frequencia in frequencias:
            if frequencia == "A":
                faltas += 1
            if frequencia == "M":
                continue
            total += 1
                
        if not calcular_frequencia(total, faltas):
            alunos_freq_baixa.append(aluno)
        
    print(" ".join(alunos_freq_baixa) if len(alunos_freq_baixa) > 0 else "")